qiskit/primitives/base/base_estimator.py

```diff
@@ -258,3 +260,42 @@ def parameters(self) -> tuple[ParameterView, ...]:
             Parameters, where ``parameters[i][j]`` is the j-th parameter of the i-th circuit.
         """
         return tuple(self._parameters)
+
+
+BaseEstimator = BaseEstimatorV1
+
+
+class BaseEstimatorV2(BasePrimitiveV2, Generic[T]):
+    """Estimator base class version 2.
+
+    Estimator estimates expectation values of quantum circuits and observables.
+    """
+
+    def __init__(self, options: Optional[BasePrimitiveOptionsLike]):
+        super().__init__(options=options)
+
+    def run(self, tasks: EstimatorTaskLike | Iterable[EstimatorTaskLike]) -> T:
+        """Run the tasks of the estimation of expectation value(s).
+
+        Args:
+            tasks: a tasklike object. Typically, list of tuple
+                ``(QuantumCircuit, observables, parameter_values)``
+
+        Returns:
+            The job object of Estimator's Result.
+        """
+        if isinstance(tasks, EstimatorTask):
+            tasks = [tasks]
+        elif isinstance(tasks, tuple) and isinstance(tasks[0], QuantumCircuit):
+            tasks = [EstimatorTask.coerce(tasks)]
+        elif tasks is not EstimatorTask:
```

Comment:
Do you intend something like `elif isintance(tasks, list)`? I don't get why you check `is not`.

Reply:
This is because it would be complex with tuple branches above, but it is true that it is better to check properly and raise an exception.

Reply:
updated